invenio_app_rdm/records_ui/views/filters.py

```diff
@@ -177,26 +177,14 @@ def custom_fields_search(field, field_value):
     if not namespaces.get(namespace):
         return None
 
-    namespace_string = "\:".join(namespace_array)
-    return url_for(
-        "invenio_search_ui.search", q=f"custom_fields.{namespace_string}:{field_value}"
-    )
-
-
-def custom_vocab_fields_search(field, field_value, sub_field):
-    """Get custom field search url for vocabulary values."""
-    namespace_array = field.split(":")
-    namespace = namespace_array[0]
-    namespaces = current_app.config.get("RDM_NAMESPACES")
-
-    if not namespaces.get(namespace):
-        return None
-
-    locale = get_locale()
-    if not locale:
-        locale = current_app.config.get("BABEL_DEFAULT_LOCALE", "en")
-
-    namespace_string = "\:".join(namespace_array) + f".{sub_field}.{locale}"
+    localised_title = field_cfg.get("locale")
+    if localised_title:
+        locale = get_locale()
+        if not locale:
+            locale = current_app.config.get("BABEL_DEFAULT_LOCALE", "en")
+        namespace_string = "\:".join(namespace_array) + f".{localised_title}.{locale}"
```

Comment:
nit: I have some trouble imagining what the final value will be here, maybe add a comment above with an example? Or if it's too much it can also go in the function docstring.